File: AuctionWebsite/BidService/src/main/java/com/bidservice/service/BidService.java
package com.bidservice.service;

import java.util.List;

import com.bidservice.entity.Bid;

public interface BidService {
	Bid placeBid(Bid bid);

	List<Bid> getBidsByAuction(int auctionId);

	double getHighestBid(int auctionId);

	Bid getHighestBidder(int auctionId);
}

File: auction/src/services/bidService.js
import apiClient from '../api/apiClient';

const bidService = {
    placeBid: (bidData) => apiClient.post('/bids', bidData),
    getBidsByAuction: (auctionId) => apiClient.get(`/bids/${auctionId}`),
    getHighestBid: (auctionId) => apiClient.get('/bids/highest', { params: { auctionId } }),
    getHighestBidder: (auctionId) => apiClient.get('/bids', { params: { auctionId } })
};

export default bidService;
